Phillip Smith

smithpb opened the pull request:
Submission of project for Front-end Project Week.

@Lndubose 

## Trello Set Up:

- [x] Create a Trello account
- [x] Create a new board called "Lambda Notes - {Your Name}"
- [x] Create lists titled `backlog`,`To Do`, `In Progress`, and `Done`
- [x] Fill in the `To Do` list with the MVP features listed below
- [x] Fill in the `backlog` list with all the extra features listed below
- [x] Share your board with the project manager that has been assigned to you. If you have not been assigned yet, reach out to your lead PM for guidance
- [x] Add your Trello URL to your project's README.md file. Commit the change, push it to your repository & submit a pull request

## MVP Features:

- [x] Display a list of notes
- [x] Create a note with a title and content
- [x] View an existing note
- [x] Edit an existing note
- [x] Delete an existing note
- [x] Wire up your static React App to our Notes API.
  - Now that you've completed the MVP for this project, we want to work with some data that will be persistent across a server. [We have built a Notes API](/NOTES.md) for you. You'll find information about this server in the NOTES.md file.

smithpb committed:
Add Trello and React environment

smithpb committed:
Add axios and display note titles

I added a basic get request to the server and displayed a list of the note titles to the screen.

smithpb committed:
Add react-router and Route to NoteList

I included react-router to the application and set up the first route to NoteList. Next I want to get routes set up to all the other pages for the app starting with individual note cards.

smithpb committed:
Add individual note page

I added a new component NoteCard.js and changed Note.js to be what displays the single note page. Right now it just shows title and content with no styling.

smithpb committed:
Add ability to create and delete notes

smithpb committed:
Add Navigation links

I added two links that should always show that will take the user to the list of notes and the page to create a new note.

smithpb committed:
Complete delete function

I went through and finished the function to delete a note. Now when a note is deleted, the app returns to the list view with the deleted item removed.

smithpb committed:
Create update form

I added a Route to the NoteForm to handle updating a specific note. Right now, it just populates the form with the information and no post request can be made yet.

smithpb committed:
Add ability to edit a note

I finished adding a put request that allows the user to edit the title and the content of each note.

JohnJustinn reviewed (approved):
Looks like you've got a great start!

smithpb committed:
Install styled-components

smithpb committed:
Style list of notes

smithpb committed:
Style create note form

smithpb committed:
Style individual note page

With this complete, everything should be properly styled as per the provided design files.

smithpb committed:
Add delete confirmation

I forgot to include a modal that has the user confirm the deletion of a note. I also cleaned up a bit of the code by removing some old commented out code.

smithpb committed:
Add redux to package

smithpb committed:
Set up basic redux structure

I went through and started to refactor things to use redux. Created basic actions and reducers and sent state to a few major components.

smithpb committed:
Fix update form to redirect

smithpb committed:
Add drag and drop dependencies

I added Muuri and React-Muuri to handle drap and drop functionality. This required the addition of HammerJS and Web-Animations as well.

smithpb committed:
Initial Muuri setup

I added the basic requirements to get the Muuri Library to handle drag sorting.  The syles are messed up as a result right now and there are several bugs I'll need to work out.

smithpb committed:
Fix basic styling

I fixed the list page to display properly. I changed the Link to only work when clicking the Note Title cause there was an issue were the click event would fire on dragdrop and I couldn't find any useful solution to the problem. Now anywhere but the title will allow the user to drag the note. Having trouble sorting a bug that stacks all the notes on top of each other when the app initializes and a few styling problems when dealing with the actively dragging item.

smithpb committed:
Fix note stacking bug

I fixed the issue where all the notes would be stack ontop of one another when the app initialized. Also adjusted the CSS to allow the dragged note to appear on top of the list though z-indexing.

smithpb committed:
Allow drag and drop to change state

After going over the very limited documentation, I figured out how to wire up react-muuri to link DOM changes to the store.

smithpb committed:
Fix bugs

I fixed a few issues that was preventing click events from triggering on cards in the list. I had to put in a distance delay for registering the drag using the dragStartPredicate option from muuri. Also fixed another issue where a newly created note would be rendered to the list incorrectly.

smithpb committed:
Link update request to state

First try

smithpb committed:
Link delete request to state

With this commit, all CRUD methods are linked to the store. Basic dragging and dropping is complete. I may look into using drag and drop to delete an item when in list view.

smithpb committed:
Add search functionality

I added a basic search feature. Right now, found items just turn red. I will work on making it more interesting in the future.

smithpb committed:
Add react-markdown

I included react-markdown to the node packages. Now the Note.js component supports Markdown. Next I will try to include limited markdown support for the NoteCard rendered to the NoteList.

smithpb committed:
Add markdown to NoteCard.js

smithpb committed:
Add ability to clear searches

I added an action that allows the user to clear search results and return to a default state in the NoteList.

smithpb committed:
Begin refactoring styles

smithpb committed:
Add card flip

I continued to refactor style to be centralized in one file. I also added a button that will flip each card in preparation to add delete function to NoteList.

smithpb committed:
Add mass delete to NoteList

I made it so the user can select notes while in list view and delete multiple notes at once. Results to synch to the DOM yet. I may have to move a few things around.

smithpb committed:
Fix display on mass delete

I fixed it so that the display changes when the user deletes multiple notes from the list.

smithpb committed:
Refine styling

I fixed a few styling issues that were bothering me. The delete button on each card is now an x box. When the user selects notes to delete, a trash can appears to confirm the deletion. There's an issue with clicking on the delete x when a note is not returned in the user search. The back of the card does not display correctly. More tinkering is needed to figure out what is going wrong there.

smithpb committed:
Move styles to separate files

smithpb committed:
Fix update note and clean component files

I finally fixed the note update form to redirect back to the Note page with the updated information. It was simply adding the componentDidUpdate lifecycle method. I also moved the styles form the components to the dedicated style files.

smithpb committed:
Clean code

I went througn everything to clean up the presentation of the code. Included comments where I felt they were necessary. Removed old commented out code.
